UnityProject/Assets/Scripts/Items/Food/Torch.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using Mirror;
+using System.Collections;
+using UnityEngine;
+using Random = UnityEngine.Random;
+using Items;
+
+namespace items
+{
+    /// <summary>
+    /// Lightable objects that can be used as lighters
+    /// </summary>
+    public class Torch : NetworkBehaviour, ICheckedInteractable<HandApply>,
+        ICheckedInteractable<InventoryApply>, IServerDespawn
+    {
+		private const int DEFAULT_SPRITE = 0;
+		private const int LIT_SPRITE = 1;
+
+		[SerializeField]
+		[Tooltip("Object to spawn after torch burnt")]
+		private GameObject burntOutPrefab = null;
+
+		[SerializeField]
+		[Tooltip("Time after torch will destroy and spawn burnt remains")]
+		private float burnTimeSeconds = 30;
+
+		[SyncVar]
+		private bool isLit = false;
+
+		private SpriteHandler spriteHandler;
+		private FireSource fireSource;
+		private Pickupable pickupable;
+
+		private void Awake()
+		{
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+			fireSource = GetComponent<FireSource>();
+			pickupable = GetComponent<Pickupable>();
+		}
+
+		public void OnDespawnServer(DespawnInfo info)
+		{
+			ServerChangeLit(false);
+		}
+
+		#region Interactions
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			// standard validation for interaction and harm intent
+			if (!DefaultWillInteract.Default(interaction, side))
+			{
+				return false;
+			}
+
+			ItemAttributesV2 attr = interaction.TargetObject.GetComponent<ItemAttributesV2>();
+			return attr.HasTrait(CommonTraits.Instance.LightableSurface) && interaction.Intent == Intent.Harm;
+		}
+
+		public bool WillInteract(InventoryApply interaction, NetworkSide side)
+		{
+			// standard validation for interaction
+			if (DefaultWillInteract.Default(interaction, side) == false || !interaction.TargetObject)
+			{
+				return false;
+			}
+
+			ItemAttributesV2 attr = interaction.TargetObject.GetComponent<ItemAttributesV2>();
+			return attr.HasTrait(CommonTraits.Instance.LightableSurface) && interaction.Intent == Intent.Harm;
+		}
+
+		public void ServerPerformInteraction(InventoryApply interaction)
+		{
+			TryLightByObject();
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			TryLightByObject();
+		}
+
+		#endregion
+
+		private void ServerChangeLit(bool isLitNow)
+		{
+			if (spriteHandler)
+			{
+				var newSpriteID = isLitNow ? LIT_SPRITE : DEFAULT_SPRITE;
+				spriteHandler.ChangeSprite(newSpriteID);
+			}
+
+			// toggle flame from match
+			if (fireSource)
+			{
+				fireSource.IsBurning = isLitNow;
+			}
+
+			if (isLitNow)
+			{
+				StartCoroutine(FireRoutine());
+			}
+		}
+
+		private void TryLightByObject()
+		{
+			if (!isLit)
+			{
+				ServerChangeLit(true);
+			}
+
+			return;
```

Comment:
no need to return at the end of a void method. BTW the name of the method implies it will make some kind of validation or that the action might fail, change it to  ``LightByObject``.